be/src/agent/heartbeat_server.cpp

```diff
@@ -150,6 +153,50 @@ void HeartbeatServer::heartbeat(THeartbeatResult& heartbeat_result, const TMaste
     }
 }
 
+std::string HeartbeatServer::print_master_info(const TMasterInfo& master_info) const {
+    using ::apache::thrift::to_string;
```

Comment:
or here make a copy of master_info, clear the token and then use thrift generated TMasterInfo::printTo() to generate the string.